src/webots/nodes/WbBackground.cpp

```diff
@@ -275,10 +297,9 @@ void WbBackground::applySkyBoxToWren() {
 
   QString textureUrls[6];
   QVector<float *> hdrImageData;
-  QVector<QImage *> regularImageData;
-
   // 1. Load the background.
   mCubeMapTexture = wr_texture_cubemap_new();
+  QImage image[6];
```

Comment:
This is because the texture image data should be persistent until the WREN texture cubemap is fully setup.
That's because `wr_texture_cubemap_set_data` is not making a copy of the image data, but re-using the one from the QImage object, so the `QImage` object must remain persistent until the cubmap construction is complete.
I added a comment in the code about it.

Reply:
Ok, thank you for the explanation and comment.